homeassistant/components/aten_pe/switch.py

```diff
@@ -0,0 +1,110 @@
+"""The ATEN PE switch component."""
+
+import logging
+
+from atenpdu import AtenPE
+import voluptuous as vol
+
+from homeassistant.components.switch import (
+    DEVICE_CLASS_OUTLET,
+    PLATFORM_SCHEMA,
+    SwitchDevice,
+)
+from homeassistant.const import CONF_HOST, CONF_PORT, CONF_USERNAME
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_AUTH_KEY = "auth_key"
+CONF_COMMUNITY = "community"
+CONF_PRIV_KEY = "priv_key"
+DEFAULT_COMMUNITY = "private"
+DEFAULT_PORT = "161"
+DEFAULT_USERNAME = "administrator"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+        vol.Optional(CONF_COMMUNITY, default=DEFAULT_COMMUNITY): cv.string,
+        vol.Optional(CONF_USERNAME, default=DEFAULT_USERNAME): cv.string,
+        vol.Optional(CONF_AUTH_KEY): cv.string,
+        vol.Optional(CONF_PRIV_KEY): cv.string,
+    }
+)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up the ATEN PE switch."""
+    dev = AtenPE(
```

Comment:
We should check that the connection including credentials is working here somehow. If it's not working we should take appropriate action, eg `raise PlatformNotReady` or log error and `return`.

Reply:
I briefly looked at the library and it doesn't seem to use the asyncio api of pysnmp. Please confirm what's the case. This will affects how we should do things here in home assistant.

Reply:
You're right. I've updated both the library and the component to use asyncio now.